asciidoctorj-core/src/main/java/org/asciidoctor/extension/Processor.java

```diff
@@ -16,25 +17,29 @@
 
 public class Processor {
 
-	protected RubyHash config;
-    protected Ruby rubyRuntime;
+	protected Map<String, Object> config;
 
     public Processor(Map<String, Object> config) {
-        this.rubyRuntime = JRubyRuntimeContext.get();
-        this.config = RubyHashUtil.convertMapToRubyHashWithSymbols(rubyRuntime, config);
+        this.config = config;
     }
 
-    public void update_config(Map<Object, Object> config) {
+    public void update_config(Map<String, Object> config) {
     	this.config.putAll(config);
     }
     
-    public Map<Object, Object> getConfig() {
+    public Map<String, Object> getConfig() {
     	return this.config;
     }
 
-    public RubyHash setConfig(Map<String, Object> config) {
-        this.config = RubyHashUtil.convertMapToRubyHashWithSymbols(rubyRuntime, config);
```

Comment:
I don't think I had any intention for the config to be modified after the initializer is called, so I think it's safe to disable this access for now (at least until we have a case).

The InlineMacroProcessor on the Ruby side is still very buggy in the construction phase. That's something I need to deal with in 1.5.3.